android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java

```diff
@@ -3,44 +3,192 @@
+                Settings.System.canWrite(getReactApplicationContext());
+
+        payload.putBoolean("hasPermission", hasPermission);
+
+        getReactApplicationContext()
+                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                .emit(PERMISSION_EVENT_NAME, payload);
+    }
+
+    /**
+     * Returns whether the device has granted the application permission to write settings.
+     *
+     * @return True if WRITE_SETTINGS are granted.
+     */
+    private boolean hasSettingsPermission() {
+        return Build.VERSION.SDK_INT < Build.VERSION_CODES.M ||
+                (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+                Settings.System.canWrite(getReactApplicationContext()));
+    }
+
+    /**
+     * Invokes the request permission activity to request access for WRITE_SETTINGS.
+     */
+    private void requestSettingsPermission() {
+        Context context = getReactApplicationContext();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !Settings.System.canWrite(context)) {
+            Intent intent = new Intent(
+                    Settings.ACTION_MANAGE_WRITE_SETTINGS,
+                    Uri.parse("package:" + context.getPackageName())
+            );
+            mActivity.startActivityForResult(intent, mWriteSettingsRequestCode);
+        }
+    }
+
+    /**
+     * Gets the brightness level of the device settings.
+     *
+     * @return The brightness level.
+     */
+    private Integer getSystemBrightness() {
+        Integer brightness;
+        try {
+            brightness = Settings.System.getInt(
+                    getReactApplicationContext().getContentResolver(),
+                    Settings.System.SCREEN_BRIGHTNESS
+            );
+        } catch (Settings.SettingNotFoundException e) {
+            brightness = null;
+        }
+        return brightness;
+    }
+
+    /**
+     * Sets the brightness level to the device settings.
+     *
+     * @param brightness The brightness level between 0-255.
+     * @return True if the brightness has been set.
+     */
+    private boolean setSystemBrightness(int brightness) {
+        if (hasSettingsPermission()) {
+            // ensure brightness is bound between range 0-255
+            brightness = Math.max(BRIGHTNESS_MIN, Math.min(brightness, BRIGHTNESS_MAX));
+            Settings.System.putInt(
+                    getReactApplicationContext().getContentResolver(),
+                    Settings.System.SCREEN_BRIGHTNESS,
+                    brightness
+            );
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Determines if the application has been granted WRITE_SETTINGS permissions.
+     * This method is callable from the JS context.
+     *
+     * @param promise A promise resolving if the application has WRITE_SETTINGS granted
+     */
+    @ReactMethod
+    public void hasPermission(final Promise promise) {
```

Comment:
Yes - `@ReactMethods` are exportable for the JS context.

We basically have two public APIs - native side & JS.

Typically any public method on the native module will be a parent override for lifecycle hooks, or if the native module should be used by something in the activity.

Otherwise the other public API is for the JS context